proposals/NNNN-allow-let-property-wrapper.md

```diff
@@ -0,0 +1,239 @@
+```swift
+struct ContentView: View {
+  @ScaledMetric let imageSize = 10
+
+  var body: some View {
+    Image(systemName: "heart.fill")
+    .resizable()
+    .frame(width: imageSize, height: imageSize)
+  }
+}
+```
+
+Similarly, other SwiftUI property wrappers could be `let` declared when they do not require more than a single initialization.
+
+## Alternatives considered
+
+### @State and @Binding property wrappers with `nonmutating set`
+
+Currently, SwiftUI offers two property wrappers,  [`@State`](https://developer.apple.com/documentation/swiftui/state/wrappedvalue) and [`@Binding`](https://developer.apple.com/documentation/swiftui/binding/wrappedvalue), that have a `wrappedValue` property with a `nonmutating set`. This allows the backing types to preserve the reference semantics of the `wrappedValue` implementation.
+
+A `var` declared `@State` property currently generates a nonmutating set that reflects `@State`'s reference based storage:
+
+```swift
+@State var weekday: String = "Monday"
+
+private var _weekday: State<String> = State<String>(wrappedValue: "Monday")
+var weekday : String {
+  get { 
+    return _weekday.wrappedValue 
+  }
+  nonmutating set {
+    self._weekday.wrappedValue = value
+  }
+  nonmutating _modify { 
+    yield () 
+  }
+}
+```
+
+Declaring the `@State` property with a `let` will remove the nonmutating setter:
+
+```swift
+@State let weekday: String = "Monday"
+
+private let _weekday: State<String> = State<String>(wrappedValue: "Monday")
+var weekday : String {
+  get { 
+    return _weekday.wrappedValue 
+  }
+}
+```
+
+[There is an argument to be made](https://forums.swift.org/t/pitch-allow-property-wrappers-on-let-declarations/61750/20) in favor of retaining the `nonmutating set` for let declared `@State` and `@Binding` to signify their reference based storage. Afterall, a property wrapper's `wrappedValue` could be assigned via the `_weekday` backing property, even if `weekday` was marked with a `let` instead of a `var`.
+
+However, we have elected to remove the `nonmutating set` for `let` declared `@State` or `@Binding` to maintain language consistency. Since property wrapper is just a struct or class with an attribute and a `wrappedValue`, a struct with a backing type of a class would not generate a nonmutating setter for its instances. For example:
+
+```swift
+class A {
+  init () {}
+}
+
+struct B {
+  var num: A
+
+  init(n: A) {
+    self.num = n
+  }
+}
+
+struct Test {
+  let b: B // note: change 'let' to 'var' to make it mutable
+  
+  var test: A {
+    get { b.num }
+    nonmutating set { b.num = newValue } // error: cannot assign to property: 'b' is a 'let' constant
+  }
+}
+```
+
+Should SwiftUI or general property wrapper usage evolve in a different direction in the future, this decision can be reconsidered to propagate backing type traits to property wrapper declarations.
```

Comment:
```suggestion
Should property wrapper usage evolve in a different direction in the future, this decision can be reconsidered.
```

I don't know what "backing type traits" are but it doesn't seem necessary to understand what's being described.